frontend/src/scenes/instance/Licenses/index.tsx

```diff
@@ -8,66 +9,175 @@ import { PageHeader } from 'lib/components/PageHeader'
+        .sort((a, b) => a.id - b.id)
+    const willDeleteProjects = !hasAnotherValidLicense && nonDemoProjects.slice(1, nonDemoProjects.length).length > 0
+
+    return (
+        <LemonModal
+            visible={true}
+            onCancel={onCancel}
+            title="Are you sure you want to cancel your license?"
+            footer={
+                <>
+                    <LemonButton
+                        form="new-dashboard-form"
+                        type="secondary"
+                        data-attr="cancel-license-cancel"
+                        style={{ marginRight: '0.5rem' }}
+                        onClick={onCancel}
+                    >
+                        Cancel
+                    </LemonButton>
+                    <LemonButton data-attr="cancel-license" type="primary" status="danger" onClick={onOk}>
+                        {willDeleteProjects ? (
+                            <>Deactivate license & delete {nonDemoProjects.length} project(s)</>
+                        ) : (
+                            'Deactivate license'
+                        )}
+                    </LemonButton>
+                </>
+            }
+        >
+            <ul>
+                {!hasAnotherValidLicense ? (
+                    <li>
+                        You will <strong>IMMEDIATELY</strong> lose access to all premium features such as{' '}
+                        <strong>multiple projects</strong>, <strong>single sign on</strong>,{' '}
+                        <strong>group analytics</strong>, <strong>multivariate feature flags</strong> and many more.
+                    </li>
+                ) : (
+                    <li>
+                        You will <strong>keep</strong> all premium features available with your other valid license.
+                    </li>
+                )}
+                {willDeleteProjects && (
+                    <li>
+                        We will <strong style={{ color: 'var(--danger)' }}>DELETE</strong> the following projects:
+                        <ul>
+                            {nonDemoProjects.map((team: TeamType) => (
+                                <li key={team.id}>
+                                    <strong>{team.name}</strong>
+                                </li>
+                            ))}
+                        </ul>
+                        To keep one of these projects instead, remove all other projects first.
+                    </li>
+                )}
+                <li>You will immediately be billed for usage in the current period, if any.</li>
+            </ul>
+        </LemonModal>
+    )
+}
+
+export function Licenses(): JSX.Element {
+    const [form] = Form.useForm()
+    const { licenses, licensesLoading, error } = useValues(licenseLogic)
+    const { createLicense, deleteLicense } = useActions(licenseLogic)
+    const [showConfirmCancel, setShowConfirmCancel] = useState(undefined as LicenseType | undefined)
```

Comment:
This should really be in `licenseLogic`, which would for instance allow easy filtering the license being canceled out of `hasAnotherValidLicense`.

Reply:
I'm not sure I understand the second point here? If I moved it into the logic i'd still be passing `license` around to show the cancel modal for the correct onel

Reply:
Well, yes, but via `useValues`, which would consolidate state.